### PR TITLE
Allow both client & server to operate in dual mode to allow ipv4 or i…

### PR DESCRIPTION
**Pull request by zentron**
…pv6 connections

### CODE REVIEW

**Comment by zentron** [comment]
What are thoughts on this approach? Should we enable dual mode like this, or provide a config setting to allow ipv6 when its needed?

**Comment by zentron** [comment]
Relates to https://github.com/OctopusDeploy/Issues/issues/1890

File: source/Halibut.SampleServer/Program.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static void Main(string[] args)
             Console.Title = "Halibut Server";
             var certificate = new X509Certificate2("HalibutServer.pfx");
 
-            var endPoint = new IPEndPoint(IPAddress.Any, 8433);
+            var endPoint = new IPEndPoint(IPAddress.IPv6Any, 8433);
```
Comment (PaulStovell) [inline source/Halibut.SampleServer/Program.cs]: It's so weird that IPv6Any also listens on v4

Comment (zentron) [inline source/Halibut.SampleServer/Program.cs]: It in-and-of itself doesnt.. It also requires
`listener.Server.DualMode = (endPoint.Address.AddressFamily == AddressFamily.InterNetworkV6);`
below

**Comment by zentron** [comment]
I might add that this works for test clients, im currently in the process of testing against Octopus Deploy itself

File: source/Halibut.SampleServer/Program.cs (hunk 1 above)
Comment (zentron) [inline source/Halibut.SampleServer/Program.cs]: And apparently even accessing `DualMode` if the listener is only ipv4 throws an exception.

**Pull request merged**